docs/DeveloperGuide.md

```diff
@@ -1,29 +1,693 @@
+![Sequence diagram for List Expense Feature](diagrams/ExpenseList_SequenceDiagram.png)
+
+
+### Currency Converter feature
+The Currency Converter Feature allows users to convert the currency of expenses and savings. This feature is facilitated by the `ChangeCurrencyCommand` class, initialized by the `Parser` class with `CurrencyConverter`, `ExpenseList`, and `SavingList` objects, alongside the `newCurrency` to convert to. The importance of these class attributes is as follows:
+
+| Class Attribute   | Variable Type          | Relevance                                                   |
+|-------------------|------------------------|-------------------------------------------------------------|
+| currencyConverter | CurrencyConverter      | The object responsible for currency conversion calculations |
+| expenseList       | ExpenseList            | Contains the expenses whose currency will be converted      |
+| savingList        | SavingList             | Contains the savings whose currency will be converted       |
+| newCurrency       | Currency               | The new currency to which the amounts will be converted     | 
+| exchangeRates     | Map<Currency, Double>  | Stores exchange rates with currencies as keys               |
+
+When `BudgetBuddy` calls `command.execute()`, `ChangeCurrencyCommand` employs the following methods from `CurrencyConverter` to convert the currency of all financial records:
+
+| Method                   | Return Type | Relevance                                                                 |
+|--------------------------|-------------|---------------------------------------------------------------------------|
+| convertExpenseCurrency() | void        | Converts the currency of each `Expense` object to `newCurrency`           |
+| convertSavingCurrency()  | void        | Converts the currency of each `Saving` object to `newCurrency`            |
+| convertAmount()          | double      | Converts an amount from one currency to another using the exchange rates  |
+
+The Currency Converter feature also includes a mechanism for managing a default currency across the application, facilitated by the `DefaultCurrency` class. This enhancement allows for seamless conversion of financial records to a user-specified default currency.
+
+The `DefaultCurrency` class is designed to maintain and update the application-wide default currency setting. It provides static methods to get and set the default currency:
+
+| Method               | Return Type | Relevance                                                    |
+|----------------------|-------------|--------------------------------------------------------------|
+| getDefaultCurrency   | Currency    | Retrieves the current default currency for the application   |
+| setDefaultCurrency   | void        | Updates the default currency to a new value                  |
+
+
+Here's the step-by-step process when the user uses the Currency Converter feature:
```

Comment:
I think since you have the sequence diagrams, I feel it is not necessary to explain your commands again